### DISCUSSION TITLE
ARM64 Support

### DISCUSSION THREAD

**Discussion by brianriggan** [derceg/explorerplusplus#369] [General]
How difficult would it be to support ARM64 for this project? I have tried implementing it myself but I do not have a ton of C++ experience and was not able to get it to compile.

**Comment by derceg** [comment]
It looks like everything compiles cleanly for ARM64, with some minor changes. The blocker at the moment is Boost. The NuGet package that's been used (https://github.com/sergey-shandar/getboost) includes pre-built binaries. There are no binaries for ARM64, however, so it's not possible to target that platform.

I'll fix that by switching to vcpkg for dependency management, which will also bring some other improvements. I have that working locally at the moment, including a clean build for ARM64, so there should be ARM64 builds in the near future.

Just a note, though, that I won't be able to test the ARM64 builds at all, since I don't have an ARM64 device.

**Reply by brianriggan** [reply]
This is excellent! I very much appreciate the support and I would be glad to test this on physical ARM64 hardware, I have 2 different ARM machines that I am currently working with. Thanks for your help.

**Comment by derceg** [comment]
There's now an ARM64 build available on AppVeyor. The following link should let you download it:

https://ci.appveyor.com/api/projects/derceg/explorerplusplus/artifacts/explorerpp_arm64.zip?branch=master&pr=false&job=Platform%3A%20ARM64

Let me know if it works. Provided it does, I'll add a link to it in the project readme and on the downloads page of the website.

**Reply by brianriggan** [reply]
Hi David, I tested this on Windows 11 ARM64 and Windows PE ARM64 on two different systems. Everything appears to be working as expected on Windows 11 ARM64 and mostly everything appears to be working on Windows PE ARM64. The only issue that I saw was an error when closing the application "The thread tried to read from or write to a virtual address for which it does not have the appropriate access." To try and rule things out I tested the x64 version of the tool in Windows PE x64 and it works as expected. (No crash when closing). I have attached the .dmp files I captured. Thank you again for your support on this.  
[Explorer++1.4.0.1957-28112022-151850.zip](https://github.com/derceg/explorerplusplus/files/10105543/Explorer%2B%2B1.4.0.1957-28112022-151850.zip)
